browser/brave_webcompat_reporter/webcompat_reporter_service.cc

```diff
@@ -0,0 +1,102 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_webcompat_reporter/webcompat_reporter_service.h"
+
+#include <string>
+
+#include "net/base/load_flags.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "base/task/post_task.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "chrome/browser/browser_process.h"
+#include "base/json/json_writer.h"
+#include "base/environment.h"
+#include "brave/components/brave_referrals/browser/brave_referrals_service.h"
+#include "brave/common/network_constants.h"
+
+namespace brave {
+
+BraveWebcompatReporterService::BraveWebcompatReporterService() {}
+
+BraveWebcompatReporterService::~BraveWebcompatReporterService() {}
+
+void BraveWebcompatReporterService::SubmitReport(std::string report_domain) {
+  std::string api_key = brave::GetAPIKey();
+
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  std::string upload_url;
+  env->GetVar("BRAVE_WEBCOMPAT_REPORT_ENDPOINT", &upload_url);
+  if (upload_url.empty())
+    upload_url = kBraveWebcompatReportEndpoint;
+
+  base::Value post_data_obj(base::Value::Type::DICTIONARY);
+  post_data_obj.SetKey("domain", base::Value(report_domain));
+  post_data_obj.SetKey("api_key", base::Value(api_key));
+
+  std::string post_data;
+  base::JSONWriter::Write(post_data_obj, &post_data);
+
+  base::PostTaskWithTraits(FROM_HERE, {content::BrowserThread::UI},
```

Comment:
why are we posting from UI thread to UI thread?

Reply:
I'm not super clear on what posting is, I think this is modeled from code elsewhere. What's the correct alternative?

Reply:
I think you were probably looking at the crash reporter which is running on a different thread. You're already on the UI thread so you can just call it directly.